tilecloud_chain/controller.py

```diff
@@ -279,46 +310,64 @@ def _fill_legend(
         assert gene.config_file
         config = gene.get_config(gene.config_file)
 
+    start = time.perf_counter()
+    legend_image_future = {}
+    with concurrent.futures.ThreadPoolExecutor(
+        max_workers=int(os.environ.get("TILECLOUD_CHAIN_CONCURENT_GET_LEGEND", "20"))
+    ) as executor:
+        for layer_name, layer in config.config["layers"].items():
+            if (
+                "legend_mime" in layer
+                and "legend_extension" in layer
+                and layer_name not in gene.layer_legends
+            ):
+                for zoom, resolution in enumerate(config.config["grids"][layer["grid"]]["resolutions"]):
+                    path = "/".join(
+                        [
+                            "1.0.0",
+                            layer_name,
+                            layer["wmts_style"],
+                            f"legend{zoom}.{layer['legend_extension']}",
+                        ]
+                    )
+                    legend_image_future[
+                        executor.submit(
+                            _legend_metadata,
+                            cache,
+                            layer,
+                            os.path.join(base_urls[0], server.get("static_path", "static") if server else ""),
+                            path,
+                        )
+                    ] = (layer_name, resolution)
+
+    legend_image_metadata: dict[str, dict[float, Optional[tilecloud_chain.Legend]]] = {}
+    for future in concurrent.futures.as_completed(legend_image_future):
+        layer_name, resolution = legend_image_future[future]
+        try:
+            legend_image_metadata.setdefault(layer_name, {})[resolution] = future.result()
+        except Exception as exc:
+            _LOGGER.warning(
+                "Unable to get legend image for layer '%s', resolution '%s': %s", layer_name, resolution, exc
+            )
+
+    _LOGGER.info("Get %i legend images in %s", len(legend_image_future), time.perf_counter() - start)
```

Comment:
does this need to be in `INFO` log level? seems quite verbose in a production env. It should be more `DEBUG` probably